features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/members/moderation/RoomMembersModerationView.kt

```diff
@@ -0,0 +1,318 @@
+    if (roomMember != null && actions.isNotEmpty()) {
+        val bottomSheetState = if (LocalInspectionMode.current) {
+            sheetStateForPreview()
+        } else {
+            rememberModalBottomSheetState(skipPartiallyExpanded = true)
+        }
+        ModalBottomSheet(
+            modifier = Modifier.systemBarsPadding(),
+            sheetState = bottomSheetState,
+            onDismissRequest = {
+                coroutineScope.launch {
+                    bottomSheetState.hide()
+                    onDismiss()
+                }
+            },
+        ) {
+            Column(
+                modifier = Modifier.padding(vertical = 16.dp)
+            ) {
+                Avatar(
+                    avatarData = AvatarData(
+                        id = roomMember.userId.value,
+                        name = roomMember.displayName,
+                        url = roomMember.avatarUrl,
+                        size = AvatarSize.RoomListManageUser,
+                    ),
+                    modifier = Modifier
+                        .padding(bottom = 28.dp)
+                        .align(Alignment.CenterHorizontally)
+                )
+                roomMember.displayName?.let {
+                    Text(
+                        text = it,
+                        style = ElementTheme.typography.fontHeadingLgBold,
+                        maxLines = 1,
+                        overflow = TextOverflow.Ellipsis,
+                        textAlign = TextAlign.Center,
+                        modifier = Modifier
+                            .padding(start = 16.dp, end = 16.dp, bottom = 8.dp)
+                            .fillMaxWidth()
+                    )
+                }
+                Text(
+                    text = roomMember.userId.toString(),
+                    style = ElementTheme.typography.fontBodyLgRegular,
+                    color = ElementTheme.colors.textSecondary,
+                    maxLines = 1,
+                    overflow = TextOverflow.Ellipsis,
+                    textAlign = TextAlign.Center,
+                    modifier = Modifier
+                        .padding(horizontal = 16.dp)
+                        .fillMaxWidth()
+                )
+                Spacer(modifier = Modifier.height(32.dp))
+
+                for (action in actions) {
+                    when (action) {
+                        is ModerationAction.DisplayProfile -> {
+                            ListItem(
+                                headlineContent = { Text(stringResource(R.string.screen_room_member_list_manage_member_user_info)) },
+                                leadingContent = ListItemContent.Icon(IconSource.Vector(CompoundIcons.Info())),
+                                onClick = {
+                                    coroutineScope.launch {
+                                        onActionSelected(action)
+                                        bottomSheetState.hide()
```

Comment:
If the bottomsheet is visible or not should probably be part of the `RoomMembersModerationState`

Reply:
Maybe, but that's not how M3 modal bottom sheet works. First you need to display the animation to hide, then toggle the state that makes it visible. This is the only one that precedes changing the state because it automatically moves you out of the screen, otherwise you'd have to wait until the bottom sheet is dismissed before.

Reply:
If I'm not mistaken `RoomListContextMenu` is handled like I said ^ but maybe the scenario is not the same?

Reply:
I think it is. That's super weird, every time I've tried it with just a boolean for visibility it never behaved correctly for animations 🤔 .